src/Driver/FlatFile/Driver.php

```diff
@@ -105,6 +107,8 @@ public function popMessage($queueName, $duration = 5)
                 if (@rename($queueDir.DIRECTORY_SEPARATOR.$id, $queueDir.DIRECTORY_SEPARATOR.$id.'.proceed')) {
                     return [file_get_contents($queueDir.DIRECTORY_SEPARATOR.$id.'.proceed'), $id];
                 }
+
+                return $this->process($queueDir, $id);
```

Comment:
This code just does the same as above except it throws an exception then the rename fails. What's the point?

Reply:
I guess in order to not have a runner silently fail when trying to process messages.

Personally i am okay with this, as this driver is not for production use-

Reply:
Yeah, this kind of bug is really hard to find regardless it's dev or production. I spent an hour figuring out where it fails and what was wrong because the web server and cli process were running under different users and I forgot to run `setfacl` after clearing the cache. I wish it would throw an exception rather than silently failing.

Reply:
I see. I would probably add some comments here then.

Reply:
Maybe the meaning could be reflected in `process` function name? It's private so it can be much longer, e.g `processFileOrFail`, what do you think?